chips/imxrt10xx/src/lpi2c.rs

```diff
@@ -703,9 +703,9 @@ impl<'a> Lpi2c<'a> {
 
         // setting slave address
         self.registers.mier.modify(MIER::EPIE::CLEAR);
-        self.registers
-            .mtdr
-            .write(MTDR::CMD.val(100) + MTDR::DATA.val((self.slave_address.get() << 1 + 1) as u32));
+        self.registers.mtdr.write(
+            MTDR::CMD.val(100) + MTDR::DATA.val((self.slave_address.get() << (1 + 1)) as u32),
+        );
```

Comment:
This one worries me. I assume the parentheses are just demonstrating what was already the order of precedence, but it feels sketchy and I'm not sure there isn't a bug here.

Reply:
In fact, I think there are two bugs. `100` should be `0b100` (I think), and also the last "+1" is adding to the slave address to turn it into an I2C read.

Reply:
Moved to issue #3658